### ISSUE TITLE
How to deal with multi dimensional regression problems?

### ISSUE THREAD

**Issue by visionNoob** [awslabs/autogluon#228]
Hi. Thank you for your great work! 

```
label_column = ['label1','label2','label3','label4']

predictor = task.fit(train_data=train_data, label=label_column, output_directory=dir, problem_type='regression', )

```
When I try the code above, error below is occurred. 

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-28-570edb2bca40> in <module>
      1 dir = 'mxnet_dst' # specifies folder where to store trained models
----> 2 predictor = task.fit(train_data=train_data, label=label_column, output_directory=dir, problem_type='regression', )

~/.conda/envs/py36mxnet/lib/python3.6/site-packages/autogluon/task/tabular_prediction/tabular_prediction.py in fit(train_data, label, tuning_data, output_directory, problem_type, eval_metric, hyperparameter_tune, feature_prune, auto_stack, holdout_frac, num_bagging_folds, num_bagging_sets, stack_ensemble_levels, hyperparameters, enable_fit_continuation, time_limits, num_trials, search_strategy, search_options, nthreads_per_trial, ngpus_per_trial, dist_ip_addrs, visualizer, verbosity, **kwargs)
    340                       hyperparameter_tune=hyperparameter_tune, feature_prune=feature_prune,
    341                       holdout_frac=holdout_frac, num_bagging_folds=num_bagging_folds, num_bagging_sets=num_bagging_sets, stack_ensemble_levels=stack_ensemble_levels,
--> 342                       hyperparameters=hyperparameters, time_limit=time_limits_orig, save_data=enable_fit_continuation, verbosity=verbosity)
    343         return TabularPredictor(learner=learner)

~/.conda/envs/py36mxnet/lib/python3.6/site-packages/autogluon/utils/tabular/ml/learner/default_learner.py in fit(self, X, X_test, scheduler_options, hyperparameter_tune, feature_prune, holdout_frac, num_bagging_folds, num_bagging_sets, stack_ensemble_levels, hyperparameters, time_limit, save_data, verbosity)
     52         time_preprocessing_start = time.time()
     53         logger.log(20, 'Preprocessing data ...')
---> 54         X, y, X_test, y_test, holdout_frac, num_bagging_folds = self.general_data_processing(X, X_test, holdout_frac, num_bagging_folds)
     55         time_preprocessing_end = time.time()
     56         self.time_fit_preprocessing = time_preprocessing_end - time_preprocessing_start

~/.conda/envs/py36mxnet/lib/python3.6/site-packages/autogluon/utils/tabular/ml/learner/default_learner.py in general_data_processing(self, X, X_test, holdout_frac, num_bagging_folds)
    100         # Remove all examples with missing labels from this dataset:
    101         n = len(X)
--> 102         missinglabel_indicators = X[self.label].isna().tolist()
    103         missinglabel_inds = [i for i,j in enumerate(missinglabel_indicators) if j]
    104         if len(missinglabel_inds) > 0:

~/.conda/envs/py36mxnet/lib/python3.6/site-packages/pandas/core/generic.py in __getattr__(self, name)
   5065             if self._info_axis._can_hold_identifiers_and_holds_name(name):
   5066                 return self[name]
-> 5067             return object.__getattribute__(self, name)
   5068 
   5069     def __setattr__(self, name, value):

AttributeError: 'TabularDataset' object has no attribute 'tolist'
```

How can i deal with multiple dimensional regression problems?

**Comment by jwmueller** [comment]
Sorry, predicting multiple columns at once is not yet supported.  For now, AutoGluon TabularPrediction can only predict a single column of your table.  

One crude way you can achieve for now this outlined below.
Suppose we have a table `train_data` with columns: [feat1, feat2, label1, label2], where the goal is to train models that simultaneously predict label1 and label2.  


`pred1 = task.fit(train_data.drop([label2], axis=1, inplace=False), label=label1, output_directory="pred_label1", ...)`

which will train a model to predict label1, but ensuring label2 is not used as a feature.

Next you can call:

`pred2 = task.fit(train_data.drop([label1], axis=1, inplace=False), label=label2, output_directory="pred_label2", ...)`

which will train a model to predict label2, but ensuring label1 is not used as a feature.

Here's how to get predictions for both columns on some test data.  We assume `test_data` contains neither label1 or label2 columns.
If these are present in `test_data`, then first call: `test_data.drop([label1, label2], axis=1, inplace=False)`.

`label1_predictions = pred1.predict(test_data, axis=1, inplace=False))`
`label2_predictions = pred2.predict(test_data, axis=1, inplace=False))`

This solution is a bit crude because the models ignore statistical dependence between label1 and label2 predictions, but it may still produce accurate predictions depending on your application.

A slightly more sophisticated alternative approach is detailed below.  This approach does consider dependence between the columns, but requires imposing some (arbitrary) ordering of the columns to be predicted as they are not treated symmetrically.  In this case, you should probably order these target columns such that the earlier columns are those you suspect are easier to predict (e.g. in this example, we assume label1 seems easier to predict accurately than label2):

```
label1_vals = train_data[label1]
label2_vals = train_data[label2]
train_data.drop([label1, label2], axis=1, inplace=True)

# Fit:

pred1 = task.fit(train_data, axis=1, inplace=False), label=label1, output_directory="pred_label1", ...)
train_data[label1] = label1_vals
pred2 = task.fit(train_data, axis=1, inplace=False), label=label2, output_directory="pred_label2", ...)

# Predict (we assume test_data contains neither label1 or label2):

label1_predictions = pred1.predict(test_data)
test_data[label1] = label1_predictions
label2_predictions = pred2.predict(test_data)
```

**Comment by Innixma** [comment]
Resolved in #953
